plugin/sampling/strategystore/adaptive/options.go

```diff
@@ -0,0 +1,153 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adaptive
+
+import (
+	"flag"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	targetQPS                    = "sampling.target-qps"
+	equivalenceThreshold         = "sampling.equivalence-threshold"
+	lookbackQPSCount             = "sampling.lookback-qps-count"
+	calculationInterval          = "sampling.calculation-interval"
+	lookbackInternval            = "sampling.lookback-interval"
+	delay                        = "sampling.delay"
+	defaultSamplingProbability   = "sampling.default-sampling-probability"
+	minSamplingProbability       = "sampling.min-sampling-probability"
+	lowerBoundTracesPerSecond    = "sampling.lower-bound-traces-per-second"
+	leaderLeaseRefreshInterval   = "sampling.leader-lease-refresh-interval"
+	followerLeaseRefreshInterval = "sampling.follower-lease-refresh-interval"
+
+	defaultTargetQPS                    = 1
+	defaultEquivalenceThreshold         = 0.3
+	defaultLookbackQPSCount             = 1
+	defaultCalculationInterval          = time.Minute
+	defaultLookbackInterval             = time.Minute * 10
+	defaultDelay                        = time.Minute * 2
+	defaultDefaultSamplingProbability   = 0.001
+	defaultMinSamplingProbability       = 0.00001                                      // once in 100 thousand requests
+	defaultLowerBoundTracesPerSecond    = 1.0 / (1 * float64(time.Minute/time.Second)) // once every 1 minute
+	defaultLeaderLeaseRefreshInterval   = 5 * time.Second
+	defaultFollowerLeaseRefreshInterval = 60 * time.Second
+
+	samplingLock = "sampling_lock"
+)
+
+// Options holds configuration for the adaptive sampling strategy store.
+type Options struct {
+	// TargetQPS is the target sampled qps for all operations.
+	TargetQPS float64
+
+	// QPSEquivalenceThreshold is the acceptable amount of deviation for the operation QPS from the `targetQPS`,
+	// ie. [targetQPS-equivalenceThreshold, targetQPS+equivalenceThreshold] is the acceptable targetQPS range.
+	// Increase this to reduce the amount of fluctuation in the probability calculation.
+	QPSEquivalenceThreshold float64
+
+	// LookbackQPSCount determines how many previous operation QPS are used in calculating the weighted QPS,
+	// ie. if LookbackQPSCount is 1, the only the most recent QPS will be used in calculating the weighted QPS.
+	LookbackQPSCount int
+
+	// CalculationInterval determines the interval each bucket represents, ie. if an interval is
+	// 1 minute, the bucket will contain 1 minute of throughput data for all services.
+	CalculationInterval time.Duration
+
+	// LookbackInterval is the total amount of throughput data used to calculate probabilities.
+	LookbackInterval time.Duration
```

Comment:
not sure I understand how this is different from LookbackQPSCount. 

Isn't `LookbackInterval == LookbackQPSCount * CalculationInterval` ?